### PR TITLE
Add overflow menu to projects/dataset picker activity

### PR DESCRIPTION
**Pull request by nirvn**
This PR adds a new overflow menu to the projects/datasets picker activity on Android platform. The menu holds actions to import {folder,ZIP,individual files} to the QField files directory using Android's Storage Access  Framework, which allows for nice interaction with not only system storage but also 3rd party (drive, nextcloud, etc.).

Menu is located at the top-right corner of the root screen:
image

Current actions:
image

Individual actions above have been implemented (:partying_face:), but we'll need to add a new page in our documentations on USB cable transfers (pre Android 11, and Android 11+).

### CODE REVIEW

**Comment by nirvn** [comment]
@m-kuhn , any idea why sentry-cli upload failing?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1dfc41099c339afb7fde4a592f469e3575a824c1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ab37bcb94b36e3385fc2d0d2d0076abb39168833-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ab37bcb94b36e3385fc2d0d2d0076abb39168833-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ab37bcb94b36e3385fc2d0d2d0076abb39168833-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ab37bcb94b36e3385fc2d0d2d0076abb39168833-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c27892b0259313432a58b32876dc8cfb91ca0869:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-c8cf35ea2308e797e02fe44cbe7195ecef812a10-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-c8cf35ea2308e797e02fe44cbe7195ecef812a10-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-c8cf35ea2308e797e02fe44cbe7195ecef812a10-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-c8cf35ea2308e797e02fe44cbe7195ecef812a10-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 579a0294b01736eebee44bf0f580c0d72f7e970b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ed525412e9590e75ab860007ea3ed34dbe2fc4d6-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ed525412e9590e75ab860007ea3ed34dbe2fc4d6-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ed525412e9590e75ab860007ea3ed34dbe2fc4d6-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-ed525412e9590e75ab860007ea3ed34dbe2fc4d6-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 80f3d4bd2a3244e39c8038b16a99b66b5f72c6d1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-5786fe0312166d65569ca236fe9b355c6f68603f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-5786fe0312166d65569ca236fe9b355c6f68603f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-5786fe0312166d65569ca236fe9b355c6f68603f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-5786fe0312166d65569ca236fe9b355c6f68603f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for cfbe4a78e1183b6f74b02b9777d04a2a3dc9378e:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4e637b9d95fcff478d6e7130f5d6ec005f7437b0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4e637b9d95fcff478d6e7130f5d6ec005f7437b0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4e637b9d95fcff478d6e7130f5d6ec005f7437b0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4e637b9d95fcff478d6e7130f5d6ec005f7437b0-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8c8ba527138d2de54f21be033d8b88fb7380506a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-f75d5eb2007d9fcdec292c550cbfa9649acced2b-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-f75d5eb2007d9fcdec292c550cbfa9649acced2b-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-f75d5eb2007d9fcdec292c550cbfa9649acced2b-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-f75d5eb2007d9fcdec292c550cbfa9649acced2b-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 901b5ea8d584e008a19dd033a0d739b628d13c21:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4a2f0bb3e5919873f4ee3f935ed7687ad8e8343c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4a2f0bb3e5919873f4ee3f935ed7687ad8e8343c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4a2f0bb3e5919873f4ee3f935ed7687ad8e8343c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-4a2f0bb3e5919873f4ee3f935ed7687ad8e8343c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6cda83f9eb7f24f03e63666aa7c0e407f3b9115d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-65f473d3b950d2188060049e976018b0409eed00-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-65f473d3b950d2188060049e976018b0409eed00-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-65f473d3b950d2188060049e976018b0409eed00-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-65f473d3b950d2188060049e976018b0409eed00-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 02f00858e8d5d196f8cbadf8017ce04853f03632:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-fe2bb92a3a72450a44d81bcfb43e06613a0aeb35-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-fe2bb92a3a72450a44d81bcfb43e06613a0aeb35-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-fe2bb92a3a72450a44d81bcfb43e06613a0aeb35-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-fe2bb92a3a72450a44d81bcfb43e06613a0aeb35-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for b49259f6b9eedc0dfef0772098318f74f4862dbe:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-1d94eefb5111c394937207e2f61525fe4ac98063-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-1d94eefb5111c394937207e2f61525fe4ac98063-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-1d94eefb5111c394937207e2f61525fe4ac98063-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-1d94eefb5111c394937207e2f61525fe4ac98063-x86.apk)

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 1]
```diff
@@ -54,8 +66,62 @@ public void onCreate(Bundle bundle) {
...
+            intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION |
+                            Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+            intent.addFlags(Intent.FLAG_GRANT_PREFIX_URI_PERMISSION);
+            intent.addFlags(Intent.FLAG_GRANT_PERSISTABLE_URI_PERMISSION);
+            intent.setType("application/zip");
+            startActivityForResult(intent, R.id.import_project_archive);
+            return true;
+        }
+        case R.id.usb_cable_help: {
+            String url = "https://qfield.org/docs/";
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:113]: Does this need a more precise link?

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 2]
```diff
@@ -414,10 +482,131 @@ private boolean onItemLongClick(int position) {
     protected void onActivityResult(int requestCode, int resultCode,
                                     Intent data) {
         Log.d(TAG, "onActivityResult ");
+        Log.d(TAG, "requestCode: " + resultCode);
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:485]: This looks like a duplicate of below

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java [hunk 3]
```diff
@@ -414,10 +482,131 @@ private boolean onItemLongClick(int position) {
...
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+            Intent intent = new Intent(this, QFieldProjectActivity.class);
+            intent.putExtra("path", importDatasetPath);
+            intent.putExtra("label",
+                            getString(R.string.favorites_imported_datasets));
+            startActivityForResult(intent, 123);
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:534]: Magic number?

File: platform/android/res/values/strings.xml [hunk 4]
```diff
@@ -36,4 +36,8 @@
     <string name="grant">Allow</string>
     <string name="deny_always">Deny Permanently</string>
     <string name="deny_once">Deny Once</string>
+    <string name="import_project_folder">Import Project from Folder</string>
+    <string name="import_project_archive">Import Project from ZIP</string>
+    <string name="import_dataset">Import Dataset(s)</string>
+    <string name="usb_cable_help">USB Cable Transer Help</string>
```
Comment (m-kuhn) [inline platform/android/res/values/strings.xml:42]:
```suggestion
    <string name="usb_cable_help">USB Cable Transfer Help</string>
```

File: platform/android/src/ch/opengis/qfield/QFieldUtils.java [hunk 5]
```diff
@@ -57,6 +58,28 @@ public static String getArchiveProjectName(InputStream in) {
...
+                String directoryPath = folder + file.getName() + "/";
+                new File(directoryPath).mkdir();
+                documentFileToFolder(file, directoryPath, resolver);
+            } else {
+                String filePath = folder + file.getName();
+                try {
+                    InputStream input = resolver.openInputStream(file.getUri());
+                    QFieldUtils.inputStreamToFile(input, filePath);
+                } catch (Exception e) {
+                    e.printStackTrace();
```
Comment (m-kuhn) [inline platform/android/src/ch/opengis/qfield/QFieldUtils.java:76]: Is that worth a message to the user?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6a17e2ac15c8322bba2095d87f522da174135f6a:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-7289c82361438880f7aba9b8934cc2cc75aaf05a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-7289c82361438880f7aba9b8934cc2cc75aaf05a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-7289c82361438880f7aba9b8934cc2cc75aaf05a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-7289c82361438880f7aba9b8934cc2cc75aaf05a-x86.apk)

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java (hunk 1 above)
Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:113]: @m-kuhn , yes, planning to talk to Lucie about this over the week, in the meantime, it's a "dummy" string to move things forward.

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java (hunk 2 above)
Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:485]: @m-kuhn , I had the wrong variable, fixed (resultCode -> requestCode)

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java (hunk 3 above)
Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:534]: @m-kuhn , yeah, the number we've been using forever there when opening project view/folder activity

File: platform/android/src/ch/opengis/qfield/QFieldUtils.java (hunk 5 above)
Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldUtils.java:76]: @m-kuhn , yeah, there are a few alert dialogs to put in, I'm deferring to a follow up PR to cover this and other places in the code.

File: platform/android/res/values/strings.xml (hunk 4 above)
Comment (nirvn) [reply platform/android/res/values/strings.xml:42]: @m-kuhn , well spotted.

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java (hunk 1 above)
Comment (m-kuhn) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:113]: She's on leave currently, best to talk to zsanett

Comment (nirvn) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:113]: @m-kuhn , will do.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f4581e1f6a69461bca9a0dd8265aac647c67d54d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-2cb8212da0e101fb30bb049ca0410a39cd8b1310-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-2cb8212da0e101fb30bb049ca0410a39cd8b1310-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-2cb8212da0e101fb30bb049ca0410a39cd8b1310-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2429-2cb8212da0e101fb30bb049ca0410a39cd8b1310-x86.apk)

**Comment by nirvn** [comment]
@m-kuhn , thanks for the review, all points addressed, auto-merge enabled.

**Review by m-kuhn — APPROVED** [review]

**Comment by m-kuhn** [comment]
Nice work!

File: platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java (hunk 1 above)
Comment (suricactus) [reply platform/android/src/ch/opengis/qfield/QFieldProjectActivity.java:113]: Wait until we migrate the docs I guess. @marioba

**Pull request merged**